File: TS_JT-Reader/TS_JT-Reader_Vis-Studio/TS_JT-Reader/VecU32.ts
module C_sharp_JT_Reader {
    export class VecU32 {
        private _counter: Int32 = 0;
        private _data: UInt32[];
        public setLength(size: number): void {
            this._data = new Array(size);
        }
        public ptr(): number {
            return this._counter++;
        }
    }
}

File: TS_JT-Reader/C#_JT-Reader_Vis-Studio/C sharp JT-Reader/DataType/VecU32.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using System.Windows.Forms;

namespace C_sharp_JT_Reader
{
    public class VecU32
    {
        private Int32 _counter = 0;
        private UInt32[] _data;

        // Method to set the length of the array
        public void setLength(int size)
        {
            _data = new UInt32[size];
        }

        // Method to return a pointer to the next available slot in the array
        public int ptr()
        {
            return _counter++;
        }
    }
}
